main.py

```diff
@@ -246,26 +252,44 @@ def match_flows(self, switch, args, stored_flows, many=True):
             return None
         return response
 
-    # pylint: disable=redefined-outer-name
-    # pylint: disable=too-many-branches
+    def get_action(self, switch, table_id, args, stored_flows, actions):
+        """Get actions in the matched flow"""
+        goto_table = False
+        actions_ = []
+        flow = self.match_flows(switch, table_id, args, stored_flows, False)
+        if flow and 'actions' in flow['flow']:
+            actions_ = flow['flow']['actions']
+        elif flow and 'instructions' in flow['flow']:
+            for instruction in flow['flow']['instructions']:
+                if instruction['instruction_type'] == 'apply_actions':
+                    actions_ = instruction['actions']
+                elif instruction['instruction_type'] == 'goto_table':
+                    table_id_ = instruction['table_id']
+                    if table_id < table_id_:
+                        table_id = table_id_
+                        goto_table = True
+                    else:
+                        log.error(f"A packet can only been directed to a \
```

Comment:
Notice that this isn't actually error handling, you're just logging and hoping for the best. This case is indeed exceptional, from `sdntrace_cp` perspective it's great that you're also considering this one, but it should raise and catch an exception resulting in an invalid non-terminating trace.